### PR TITLE
BFD-1748 : S3 buckets for IT tests must comply with CMS S3 security constraints

### PR DESCRIPTION
**Pull request by mackec1**
**JIRA Ticket:**
[BFD-1748](https://jira.cms.gov/browse/BFD-1748)

**User Story or Bug Summary:**
S3 buckets created as part of integration tests should comply with CMS S3 bucket security constraints:
- Not publicly accessible
- Support only TLS-enabled connection
- Data is encrypted

### What Does This PR Do?
As part of ongoing security audits, it was determined that S3 buckets that are created during Integration Tests, fail to meet CMS S3 bucket security constraints. The work associated with this ticket attempts to bring ephemeral S3 buckets created during ITs in line with CMS security policy.

When an ephemeral S3 bucket is created during integration tests, additional security constraints will be applied during bucket creation. In particular, SharedS3Utilities bucket creation includes 3 changes:

1. No public access for the created bucket
2. Enforces TLS-enabled connection to buckets
3. Data in bucket is encrypted by default.

### What Should Reviewers Watch For?
Code changes in SharedS3Utilities is concise in its functionality and commented appropriately.

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

While this PR does have security implications, they are exrremely limited in scope and pertain only to remediating an issue where S3 buckets created during integration tests, did so in an insecure manner, basically allow public access to the ephemeral buckets. This ticket closes that loop-hole.

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [X] Yes
      * [] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    4. There isn't too much of a burden on reviewers.
    5. Any problems it causes have a small "blast radius".
    6. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/java/gov/cms/bfd/pipeline/sharedutils/s3/SharedS3Utilities.java [hunk 1]
```diff
@@ -9,13 +9,7 @@
...
 import com.amazonaws.services.s3.AmazonS3;
 import com.amazonaws.services.s3.AmazonS3ClientBuilder;
-import com.amazonaws.services.s3.model.Bucket;
-import com.amazonaws.services.s3.model.GetObjectMetadataRequest;
-import com.amazonaws.services.s3.model.HeadBucketRequest;
-import com.amazonaws.services.s3.model.ListObjectsV2Request;
-import com.amazonaws.services.s3.model.ListObjectsV2Result;
-import com.amazonaws.services.s3.model.ObjectMetadata;
-import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.amazonaws.services.s3.model.*;
```
Comment (christrotteradhoc) [inline apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/java/gov/cms/bfd/pipeline/sharedutils/s3/SharedS3Utilities.java:12]: checkstyle will bark about this once it is turned on for this project. can you go back to the explicit listing?

Comment (mackec1) [reply apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/java/gov/cms/bfd/pipeline/sharedutils/s3/SharedS3Utilities.java:12]: I guess....although the number of imports will be fairly large!

Comment (mackec1) [reply apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/java/gov/cms/bfd/pipeline/sharedutils/s3/SharedS3Utilities.java]: done.

**Review by mjburling — APPROVED** [review]
I think this satisfies our requirements. Thanks, @mackec1!

**Review by christrotteradhoc — APPROVED** [review]

**Pull request merged**